File: 2024/day02/part2.py
result = 0

with open("input.txt", "r") as file:
  for line in file:
    data = list(map(int, line.rstrip().split(" ")))
    success = False
    for data in [data] + [data[:i] + data[i + 1:] for i in range(len(data))]:
      inc = all(a < b for a, b in zip(data, data[1:]))
      dec = all(a > b for a, b in zip(data, data[1:]))
      stp = all(abs(a - b) in {1, 2, 3} for a, b in zip(data, data[1:]))
      if (inc or dec) and stp:
        success = True
    if success:
      result = result + 1

print(result)

File: 2024/day03/part2.pl
$enabled = 1;
$result = 0;

open(FH, "<", "input.txt") or die $!;
while (<FH>) {
  while ($_ =~ /(do\(\))|(don't\(\))|(mul\((\d+),\s{0,}(\d+)\))/g) {
    if ($1 eq "do()") {
      $enabled = 1;
    }
    if ($2 eq "don't()") {
      $enabled = 0;
    }
    if ($enabled) {
      $result += $4 * $5;
    }
  }
}
close(FH);

print("$result\n");
